lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Context/ContextRuntime.cs

```diff
@@ -115,6 +133,17 @@ public Optional<ContextRuntime> ParentContext
             get { return _parentContext; }
         }
 
+        private static ContextRuntime GetChildContextRuntime(IInjector childServiceInjector, IConfiguration contextConfiguration, ContextRuntime parentRuntime)
+        {
+            var actualContextConfiguration = contextConfiguration as ContextConfiguration;
+            if (actualContextConfiguration == null)
+            {
+                return new ContextRuntime(childServiceInjector, contextConfiguration, Optional<ContextRuntime>.Of(parentRuntime));
+            }
+            
+            return new ContextRuntime(actualContextConfiguration.Id, childServiceInjector, contextConfiguration, Optional<ContextRuntime>.Of(parentRuntime));
```

Comment:
Yes, it will be Obsoleted in the 0.14 release, but we cannot yet remove it, so we have to support both paths prior to the 0.15 release.